### DISCUSSION TITLE
Outlining a single rep, not the others

### DISCUSSION THREAD

**Discussion by rjdirisio** [molstar/molstar#1334] [Q&A]
Hi, I was wondering if it's possible to outline (create a black border around) only a single part of an overall representation. For example, I want to outline a surface in the viewer, but not my cartoon representation. I understand that this is typically set in canvas / postprocessing and globally applies to each rep, but I was wondering if it could be controlled in a more precise way?

**Comment by rjdirisio** [comment]
This question is arising because, previous to 4.8.0, we were able to get away with having our edensity map wireframe not be outlined, even when the rep was partially transparent, but our ligand surface continued to show the outline, even though it was had transparency. This is when we had global settings in our spec that called for an outline on everything.

```tsx
      pluginRef.current.canvas3d?.setProps({
        postprocessing: {
          outline: {
            name: "on",
            params: {
              scale: 1,
              color: "#000000",
              threshold: 0.25,
            },
          },
        },
      });
    };
```

I think with https://github.com/molstar/molstar/issues/1326 and the subsequent fix https://github.com/molstar/molstar/pull/1331, the loophole we were relying on to have our wireframe rep not have an outline, but our surface to have our outline, will now be closed. In other words, I think we were relying on a bug to get the type of rep that we wanted.

**Comment by giagitom** [comment]
> In other words, I think we were relying on a bug to get the type of rep that we wanted.

Yes, I think this too sadly.


With outlines you can just use ```includeTransparent``` property to toggle outlines on transparent objects.

Depending on what you need there could be a few options.

- Use ```direct volume``` instead of ```isosurface```, since outlines are not applied on volumetric representations.

![image](https://github.com/user-attachments/assets/7e643beb-f729-4164-bb86-4ebcf4b5f336)

Or maybe apply selection marking on the surface and disable outlines on transparency 
![image](https://github.com/user-attachments/assets/fb97b880-22fc-4ed3-8410-4ace0b92d050)

**Reply by rjdirisio** [reply]
Thanks for the quick response

> Or maybe apply selection marking on the surface and disable outlines on transparency

This looks interesting! How does one do this? I'm not sure I understand what 'selection marking' means

**Reply by pechersky** [reply]
Unfortunately, selection marking isn't right, because it does the general shape outline instead of helping visualize the bumpiness of the surface.

Selection marking means, the highlight around the shape when you click on it, for example.

**Reply by pechersky** [reply]
And volumetric representations do not represent crystallographic densities that well, and obscure non-density components.

**Comment by arose** [comment]
The only way to get outlines on some parts and not other is to make the parts that should not have an outline a bit transparent and set `includeTransparent` to false in the outline params.
